wgpu-core/src/device/queue.rs

```diff
@@ -1396,6 +1396,7 @@ impl Queue {
         unsafe { self.raw().get_timestamp_period() }
     }
 
+    /// The closure is guaranteed to be called.
```

Comment:
question: Doesn't a subsequent `poll` operation have to be done for this to be guaranteed? I don't see this being called on `drop`.

This question also applies to the identical doc. comment left on `Global::buffer_map_async`.

Reply:
I think they will get called on `Queue` drop even if poll wasn't called.

Reply:
Ah, I see that `submission_closures` and `mapping_closures` are gathered in a `maintain` call in `<Queue as Drop>::drop`. 🫡 Seems in order, just needed to build a more thorough mental model.